clippy_lints/src/write.rs

```diff
@@ -187,10 +192,34 @@ declare_lint_pass!(Write => [
 ]);
 
 impl EarlyLintPass for Write {
+    fn check_item(&mut self, _: &EarlyContext<'_>, item: &Item) {
+        if let ItemKind::Impl {
+            of_trait: Some(trait_ref),
+            ..
+        } = &item.kind
+        {
+            let trait_name = trait_ref
+                .path
+                .segments
+                .iter()
+                .last()
+                .expect("path has at least one segment")
+                .ident
+                .name;
+            if trait_name == sym!(Debug) {
```

Comment:
Hm, to avoid the FN, couldn't we compare the full path here instead of just the last segment? i.e if it's `core::fmt::Debug` or `std::fmt::Debug`?

Reply:
I think we would need to turn it in a `LatePassLint` for that. I think we can live with this FN though, since it shouldn't be that common that someone defines a `Debug` trait